include/pdal/util/Extractor.hpp

```diff
@@ -240,4 +240,158 @@ class PDAL_DLL LeExtractor : public Extractor
     }
 };
 
+
+/// Stream wrapper for input of binary data that converts from little-endian
+/// to host ordering.
+class PDAL_DLL SwitchableExtractor : public Extractor
```

Comment:
The reason not to make this switchable is performance.  If there are separate classes, you eliminate all the if (little_endian) stuff.  It's in code like this where people seem to care about such things.